### PR TITLE
Avoid script location warnings on Windows

### PR DESCRIPTION
**Pull request by YannickJadoul**
This should get rid of the "WARNING: The scripts [...] are installed in '...' which is not on PATH." warnings (e.g., https://travis-ci.org/joerick/cibuildwheel/jobs/610081892#L636).

Moreover, since macOS and Windows had different ordering of the steps setting up the build environment (e.g., `CIBW_ENVIRONMENT` before or after the installation of `pip`), I've reordered the steps on Windows. Is this the correct order?

Alternatively, we could just silence the warnings with `--no-warn-script-location`, but fixing this (and keeping the warnings when they are relevant!) seems like a better solution.

Besides that, when do we want to use `simple_shell` vs. `shell`, @mayeut?

### CODE REVIEW

**Comment by YannickJadoul** [comment]
Hmmmm, the Ubuntu build on AppVeyor timed out; @joerick, the email you sent somehow didn't work to give me rights on AppVeyor. Anyhow, I'll just amend the commit without changing anything and force push again; that should start the CIs again.

**Comment by YannickJadoul** [comment]
Seem that job is timing out again. I've no idea whether that's because of the PR or because of AppVeyor (though the former would be weird, especially since the other Windows builds do succeed).

**Comment by joerick** [comment]
I've restarted it again, let's see...

**Comment by YannickJadoul** [comment]
> the email you sent somehow didn't work to give me rights on AppVeyor

(Nope, somehow still not; I think it's because my main GitHub email is a different one than in the commit messages. Wondering how I can get that to you without putting my email plainly on GitHub without the risk of being scraped and spammed. But no worries; I can perfectly survive without :) )

File: cibuildwheel/windows.py [hunk 1]
```diff
@@ -100,40 +100,44 @@ def shell(args, env=None, cwd=None):
...
             os.path.join(config_python_path, 'Scripts'),
             env['PATH']
         ])
+        # update env with results from CIBW_ENVIRONMENT
         env = environment.as_dictionary(prev_environment=env)
 
         # for the logs - check we're running the right version of python
+        shell(['where', 'python'], env=env)
         shell(['python', '--version'], env=env)
         shell(['python', '-c', '"import struct; print(struct.calcsize(\'P\') * 8)\"'], env=env)
```
Comment (mayeut) [inline cibuildwheel/windows.py]: These shall probably use the `simple_shell`

File: cibuildwheel/windows.py [hunk 2]
```diff
@@ -100,40 +100,44 @@ def shell(args, env=None, cwd=None):
...
+        # make sure pip is installed
+        if not os.path.exists(os.path.join(config_python_path, 'Scripts', 'pip.exe')):
+            simple_shell([os.path.join(config_python_path, 'python.exe'), get_pip_script], env=env)
+        assert os.path.exists(os.path.join(config_python_path, 'Scripts', 'pip.exe'))
+
         # prepare the Python environment
         shell(['python', '-m', 'pip', 'install', '--upgrade', 'pip'], env=env)
+        shell(['pip', '--version'], env=env)
         shell(['pip', 'install', '--upgrade', 'setuptools'], env=env)
         shell(['pip', 'install', 'wheel'], env=env)
```
Comment (mayeut) [inline cibuildwheel/windows.py]: These shall probably use the `simple_shell` as well.

File: cibuildwheel/windows.py (hunk 1 above)
Comment (YannickJadoul) [reply cibuildwheel/windows.py]: Thanks; that's what I thought, but they weren't before, so I wasn't sure :)

**Comment by mayeut** [comment]
> Besides that, when do we want to use `simple_shell` vs. `shell`

`shell` has a different effect on AppVeyor only where it sets-up the variables to allow C extension to build. This means `shell` shall be used where a C extension might be built. I think this goes down to: `before_build`, `build`, `install` (both), and maybe the `test_command`. All other lines can use the `simple_shell`.

I don't know if the specific script is still needed.

File: cibuildwheel/macos.py [hunk 3]
```diff
@@ -89,6 +90,7 @@ def call(args, env=None, cwd=None, shell=False):
 
         # install pip & wheel
         call(['python', get_pip_script, '--no-setuptools', '--no-wheel'], env=env)
+        assert os.path.exists(os.path.join(installation_bin_path, 'pip'))
         call(['pip', '--version'], env=env)
         call(['pip', 'install', '--upgrade', 'setuptools'], env=env)
         call(['pip', 'install', 'wheel'], env=env)
```
Comment (mayeut) [inline cibuildwheel/macos.py]: It feels weird no to upgrade `wheel` (and thus not having a single line: `call(['pip', 'install', '--upgrade', 'setuptools', 'wheel'], env=env))`

File: cibuildwheel/windows.py [hunk 4]
```diff
@@ -100,39 +100,43 @@ def shell(args, env=None, cwd=None):
...
+        assert os.path.exists(os.path.join(config_python_path, 'Scripts', 'pip.exe'))
 
         # prepare the Python environment
-        shell(['python', '-m', 'pip', 'install', '--upgrade', 'pip'], env=env)
-        shell(['pip', 'install', '--upgrade', 'setuptools'], env=env)
-        shell(['pip', 'install', 'wheel'], env=env)
+        simple_shell(['python', '-m', 'pip', 'install', '--upgrade', 'pip'], env=env)
+        simple_shell(['pip', '--version'], env=env)
+        simple_shell(['pip', 'install', '--upgrade', 'setuptools'], env=env)
+        simple_shell(['pip', 'install', 'wheel'], env=env)
```
Comment (mayeut) [inline cibuildwheel/windows.py]: Same here, It feels weird no to upgrade `wheel` (and thus not having a single line: `call(['pip', 'install', '--upgrade', 'setuptools', 'wheel'], env=env))`

Any reason you know of ?

**Comment by mayeut** [comment]
> Is this the correct order?

Seems right.

File: cibuildwheel/windows.py (hunk 4 above)
Comment (YannickJadoul) [reply cibuildwheel/windows.py]: Agreed; but I believe it's because wheel is not installed by default, so `--upgrade` wouldn't do anything.

Comment (mayeut) [reply cibuildwheel/windows.py]: This seems to work perfectly on macOS at least:
```
(.venv) Matt$ pip uninstall wheel setuptools
Uninstalling wheel-0.33.6:
  Would remove:
    /Users/Matt/Dev/pypa/setuptools/.venv/bin/wheel
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/wheel-0.33.6.dist-info/*
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/wheel/*
Proceed (y/n)? y
  Successfully uninstalled wheel-0.33.6
Uninstalling setuptools-41.6.0:
  Would remove:
    /Users/Matt/Dev/pypa/setuptools/.venv/bin/easy_install
    /Users/Matt/Dev/pypa/setuptools/.venv/bin/easy_install-3.7
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/easy_install.py
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/pkg_resources/*
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/setuptools-41.6.0.dist-info/*
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/setuptools/*
Proceed (y/n)? y
  Successfully uninstalled setuptools-41.6.0
(.venv) Matt$ pip install --upgrade wheel setuptools
Collecting wheel
  Using cached https://files.pythonhosted.org/packages/00/83/b4a77d044e78ad1a45610eb88f745be2fd2c6d658f9798a15e384b7d57c9/wheel-0.33.6-py2.py3-none-any.whl
Collecting setuptools
  Using cached https://files.pythonhosted.org/packages/d9/de/554b6310ac87c5b921bc45634b07b11394fe63bc4cb5176f5240addf18ab/setuptools-41.6.0-py2.py3-none-any.whl
Installing collected packages: wheel, setuptools
Successfully installed setuptools-41.6.0 wheel-0.33.6
```

Comment (YannickJadoul) [reply cibuildwheel/windows.py]: Yeah, it should work; I have two explanations for this:
1) One was added afterwards.
2) @joerick somehow wanted to keep those two installations separate, for some reason?

Either way, I can change this now, and see if @joerick had a reason.

File: cibuildwheel/windows.py [hunk 5]
```diff
@@ -100,39 +100,43 @@ def shell(args, env=None, cwd=None):
...
         # for the logs - check we're running the right version of python
-        shell(['python', '--version'], env=env)
-        shell(['python', '-c', '"import struct; print(struct.calcsize(\'P\') * 8)\"'], env=env)
+        simple_shell(['where', 'python'], env=env)
+        simple_shell(['python', '--version'], env=env)
+        simple_shell(['python', '-c', '"import struct; print(struct.calcsize(\'P\') * 8)\"'], env=env)
+
+        # make sure pip is installed
+        if not os.path.exists(os.path.join(config_python_path, 'Scripts', 'pip.exe')):
+            simple_shell([os.path.join(config_python_path, 'python.exe'), get_pip_script], env=env)
```
Comment (YannickJadoul) [inline cibuildwheel/windows.py]: @mayeut And here, any reason to keep the full path here, or should we shorten this to `'python'` now it's in `PATH`?

Comment (mayeut) [reply cibuildwheel/windows.py]: This can definitely be shorten to `'python'`

Comment (YannickJadoul) [reply cibuildwheel/windows.py]: Yeah, just wondering if there would be a good reason to keep it like this.

File: cibuildwheel/windows.py (hunk 4 above)
Comment (joerick) [reply cibuildwheel/windows.py]: Happy to have these both be --upgrade. Not sure why they're separate. Side tangent- I've always wanted to pin these in cibuildwheel, but it would create a maintenance burden to have to keep revving them on each release. Sure would be nice for repeatable builds though

**Comment by mayeut** [comment]
Thanks @YannickJadoul, it will be much cleaner without all the warnings in the logs.

@joerick, do you want to review this some more or can it be merged ?

**Comment by mayeut** [comment]
@YannickJadoul, if you don't have anything to add, can you merge this one ?

**Comment by YannickJadoul** [comment]
Done, thanks, @mayeut & @joerick! :-)
